dessia_common/core.py

```diff
@@ -638,10 +638,28 @@ def to_xlsx_stream(self, stream):
         writer.save_to_stream(stream)
 
     def _export_formats(self):
-        formats = [{"extension": "json", "method_name": "save_to_stream", "text": True, "args": {}},
-                   {"extension": "xlsx", "method_name": "to_xlsx_stream", "text": False, "args": {}}]
+        formats = [{"selector": "json", "extension": "json",
+                    "method_name": "save_to_stream", "text": True, "args": {}},
+                   {"selector": "xlsx", "extension": "xlsx",
+                    "method_name": "to_xlsx_stream", "text": False, "args": {}}]
         return formats
 
+    def save_export_to_file(self, selector, filepath):
+        for export_format in self._export_formats():
+            if export_format['selector'] == selector:
+                if not filepath.endswith(f".{export_format['extension']}"):
+                    filepath += f".{export_format['extension']}"
+                    print(f'Renaming filepath to {filepath}')
```

Comment:
Here is a suggestion, not sure it is better
You can replace it with : 
```
extension = f".{export_format['extension']}"
filepath += f".{extension}" * (not filepath.endswith(extension))
```

Reply:
I do really find it hard to read. The later is really straightforward and almost English, as python should be written. It literally says "If filepath does not end with extension, well, we add it".

Your solution can be sophisticated in a way, but one must just stop one or two sec to properly understand it, which is a big flaw, in my opinion